tools/generateFactories.js

```diff
@@ -13,7 +13,10 @@ export default function generateFactories(destination, babelOptions='') {
 
   let generateCompiledFile = function (file, content) {
     let outpath = path.join(destination, `${file}.js`);
-    return exec(`babel ${babelOptions} --out-file ${outpath} <<EOF\n ${content}`);
+    return fsp.writeFile(outpath + '_temp', content)
+        .then(() => exec(`babel ${babelOptions} --out-file ${outpath} ${outpath}_temp`)
+          .then(() => fsp.remove(outpath + '_temp'))
```

Comment:
I'm not a huge fan of this `then` being inside another `then`, but that is mainly my personal preference. Is there a more technical reason behind this?